QEfficient/generation/text_generation_inference.py

```diff
@@ -145,108 +145,98 @@ def cloud_ai_100_exec_kv_helper(
     stream: bool = True,
     write_io_dir: Optional[str] = None,
 ):
-    if tokenizer.padding_side != "left":
-        logger.warning("Please use padding_side='left' while initializing the tokenizer")
-        tokenizer.padding_side = "left"
+    if tokenizer.padding_side != "right":
+        logger.warning("Please use padding_side='right' while initializing the tokenizer")
+        tokenizer.padding_side = "right"
     if tokenizer.pad_token_id is None:
         tokenizer.pad_token_id = tokenizer.eos_token_id
 
     # Load QPC
     session = QAICInferenceSession(qpc, device_id, enable_debug_logs=enable_debug_logs)
 
-    # Read prompt and ctx len from session
-    prompt_len = max([x[session.binding_index_map["input_ids"]][1][1] for x in session.allowed_shapes])
-    ctx_len = session.allowed_shapes[0][session.binding_index_map["attention_mask"]][1][1]
-    if input_len is None:
-        input_len = prompt_len
-    if generation_len is None:
-        generation_len = ctx_len
-    num_chunks = -(input_len // -prompt_len)  # ceil divide without float
-    input_len = num_chunks * prompt_len  # Convert input_len to a multiple of prompt_len
-    assert input_len <= ctx_len, "input_len should be less than ctx_len"
-
     # Skip inputs/outputs
-    session.skip_buffers([x for x in session.input_names if x.startswith("past_")])
-    session.skip_buffers([x for x in session.output_names if x.endswith("_RetainedState")])
+    session.skip_buffers([x for x in session.input_names + session.output_names if x.startswith("past_")])
```

Comment:
Why this change? are we appending past_ to attention mask also?

Reply:
no, just skipping the retention `past_kvs `in the qpc execution.

Reply:
@anujgupt-github There is no attention_mask retention now, so no need of separate skipping logic for inputs and outputs.